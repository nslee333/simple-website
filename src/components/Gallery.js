import React from "react";
import { Navigation, Pagination } from "swiper";
import { Swiper, SwiperSlide } from "swiper/react";
import Image from "next/image";
import { urlFor } from "../pages";

import "swiper/swiper-bundle.min.css";
import "swiper/swiper.min.css";

export default function Gallery(images) {
  const imagesArr = images.props;
  return (
    <div className="my-10 flex justify-center ">
      <div
        className="border-white-200
                    rounded-md
                    border-8
                    xs:w-[19rem]
                    sm:w-[30rem]
                    md:w-[40rem]
                    lg:w-[50rem]"
      >
        <Swiper
          modules={[Navigation, Pagination]}
          spaceBetween={50}
          slidesPerView={1}
          navigation={true}
          pagination={{ clickable: true }}
          style={{
            "--swiper-pagination-color": "#ffffff",
            "--swiper-pagination-bullet-inactive-color": "#999999",
            "--swiper-pagination-bullet-inactive-opacity": "1",
            "--swiper-pagination-bullet-size": "10px",
            "--swiper-pagination-bottom": "25px",
            "--swiper-navigation-color": "#0",
          }}
        >
          {imagesArr ? (
            imagesArr.map((image, index) => {
              return (
                <SwiperSlide key={index}>
                  <div className="aspect" key={index}>
                    <Image
                      src={urlFor(imagesArr[index].image).url()}
                      alt={`bendjs ${index}`}
                      key={index}
                      fill
                      priority
                    />
                  </div>
                </SwiperSlide>
              );
            })
          ) : (
            <div></div>
          )}
        </Swiper>
      </div>
    </div>
  );
}
