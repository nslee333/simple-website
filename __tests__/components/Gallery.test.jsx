import { render } from "@testing-library/react";
import Gallery from "../../src/components/Gallery";
import renderer from "react-test-renderer";
test("Gallery renders without crashing", () => {
  render(<Gallery />);
});

test("Gallery renders correctly according to snapshot.", () => {
  const tree = renderer.create(<Gallery />).toJSON();

  expect(tree).toMatchSnapshot();
});
