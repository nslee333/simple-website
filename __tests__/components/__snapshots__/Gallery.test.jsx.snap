// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery renders correctly according to snapshot. 1`] = `
<div
  className="my-10 flex justify-center "
>
  <div
    className="border-white-200 rounded-md border-8 xs:w-[19rem] sm:w-[30rem] md:w-[40rem] lg:w-[50rem]"
  >
    <div
      className="swiper-container"
      modules={
        [
          {
            "create": [Function],
            "name": "navigation",
            "on": {
              "click": [Function],
              "destroy": [Function],
              "enable disable": [Function],
              "fromEdge": [Function],
              "init": [Function],
              "toEdge": [Function],
            },
            "params": {
              "navigation": {
                "disabledClass": "swiper-button-disabled",
                "hiddenClass": "swiper-button-hidden",
                "hideOnClick": false,
                "lockClass": "swiper-button-lock",
                "nextEl": null,
                "prevEl": null,
              },
            },
          },
          {
            "create": [Function],
            "name": "pagination",
            "on": {
              "activeIndexChange": [Function],
              "click": [Function],
              "destroy": [Function],
              "enable disable": [Function],
              "init": [Function],
              "slidesLengthChange": [Function],
              "snapGridLengthChange": [Function],
              "snapIndexChange": [Function],
            },
            "params": {
              "pagination": {
                "bulletActiveClass": "swiper-pagination-bullet-active",
                "bulletClass": "swiper-pagination-bullet",
                "bulletElement": "span",
                "clickable": false,
                "clickableClass": "swiper-pagination-clickable",
                "currentClass": "swiper-pagination-current",
                "dynamicBullets": false,
                "dynamicMainBullets": 1,
                "el": null,
                "formatFractionCurrent": [Function],
                "formatFractionTotal": [Function],
                "hiddenClass": "swiper-pagination-hidden",
                "hideOnClick": false,
                "lockClass": "swiper-pagination-lock",
                "modifierClass": "swiper-pagination-",
                "progressbarFillClass": "swiper-pagination-progressbar-fill",
                "progressbarOpposite": false,
                "progressbarOppositeClass": "swiper-pagination-progressbar-opposite",
                "renderBullet": null,
                "renderCustom": null,
                "renderFraction": null,
                "renderProgressbar": null,
                "totalClass": "swiper-pagination-total",
                "type": "bullets",
              },
            },
          },
        ]
      }
      style={
        {
          "--swiper-navigation-color": "#0",
          "--swiper-pagination-bottom": "25px",
          "--swiper-pagination-bullet-inactive-color": "#999999",
          "--swiper-pagination-bullet-inactive-opacity": "1",
          "--swiper-pagination-bullet-size": "10px",
          "--swiper-pagination-color": "#ffffff",
        }
      }
    >
      <div
        className="swiper-button-prev"
      />
      <div
        className="swiper-button-next"
      />
      <div
        className="swiper-pagination"
      />
      <div
        className="swiper-wrapper"
      />
      <div />
    </div>
  </div>
</div>
`;
